eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -149,7 +150,7 @@ namespace eosiosystem {
                res.ram_bytes += bytes_out;
             });
       }
-      set_resource_limits( res_itr->owner, res_itr->ram_bytes, res_itr->net_weight.amount, res_itr->cpu_weight.amount );
+      set_resource_limits( res_itr->owner, res_itr->ram_bytes + ram_gift_bytes, res_itr->net_weight.amount, res_itr->cpu_weight.amount );
```

Comment:
Similar addition of `ram_gift_bytes` needed in the other `set_resource_limits` call in `system_contract::sellram`.